[BUG] Driver 460.58 => dynamic_load_runtime.hpp:139: cudaErrorInvalidValue invalid argument

**Describe the bug**
```
0:23:56  [ERROR] testCudaAsyncMemoryResourceSize  Time elapsed: 0.002 s  <<< ERROR!
10:23:56  ai.rapids.cudf.CudfException: CUDA error at: /home/jenkins/agent/workspace/spark-rapids-jni_submodule-sync-pre_release/thirdparty/cudf/cpp/build/_deps/rmm-src/include/rmm/detail/dynamic_load_runtime.hpp:139: cudaErrorInvalidValue invalid argument
10:23:56  	at ai.rapids.cudf.Rmm.initializeInternal(Native Method)
```

per https://github.com/NVIDIA/spark-rapids-jni/issues/287#issuecomment-1140579015

**Steps/Code to reproduce bug**
On a Linux box with Driver 460.58, 

Either clone https://github.com/NVIDIA/spark-rapids-jni
and run 
```
./build/build-in-docker verify -Dtest=testCudaAsyncMemoryResourceSize
```

or clone rapidsai/cudf and run:
```
./build.sh  cudfjar
```

**Expected behavior**
The test should pass. 
This regression is presumably triggered by #1051

**Environment details (please complete the following information):**
Host Linux with  460.58 
centos7 docker based on CUDA 11.5
(cc @pxLi  for more details)
 
**Additional context**
NVIDIA/spark-rapids-jni#287

--- a/include/rmm/detail/dynamic_load_runtime.hpp
+++ b/include/rmm/detail/dynamic_load_runtime.hpp
@@ -137,15 +137,15 @@ struct async_alloc {
     int supported_handle_types_bitmask{};
 #if CUDART_VERSION >= 11030  // 11.3 introduced cudaDevAttrMemoryPoolSupportedHandleTypes
     if (cudaMemHandleTypeNone != handle_type) {
-        auto const result = cudaDeviceGetAttribute(&supported_handle_types_bitmask,
-                                                   cudaDevAttrMemoryPoolSupportedHandleTypes,
-                                                   rmm::detail::current_device().value());
+      auto const result = cudaDeviceGetAttribute(&supported_handle_types_bitmask,
+                                                 cudaDevAttrMemoryPoolSupportedHandleTypes,
+                                                 rmm::detail::current_device().value());
 
-        // Don't throw on cudaErrorInvalidValue
-        auto const unsupported_runtime = (result == cudaErrorInvalidValue);
-        if (unsupported_runtime) return false;
-        // throw any other error that may have occurred
-        RMM_CUDA_TRY(result);
+      // Don't throw on cudaErrorInvalidValue
+      auto const unsupported_runtime = (result == cudaErrorInvalidValue);
+      if (unsupported_runtime) return false;
+      // throw any other error that may have occurred
+      RMM_CUDA_TRY(result);
     }
 
 #endif

--- a/include/rmm/detail/dynamic_load_runtime.hpp
+++ b/include/rmm/detail/dynamic_load_runtime.hpp
@@ -136,15 +136,17 @@ struct async_alloc {
   {
     int supported_handle_types_bitmask{};
 #if CUDART_VERSION >= 11030  // 11.3 introduced cudaDevAttrMemoryPoolSupportedHandleTypes
-    auto const result = cudaDeviceGetAttribute(&supported_handle_types_bitmask,
-                                               cudaDevAttrMemoryPoolSupportedHandleTypes,
-                                               rmm::detail::current_device().value());
+    if (cudaMemHandleTypeNone != handle_type) {
+        auto const result = cudaDeviceGetAttribute(&supported_handle_types_bitmask,
+                                                   cudaDevAttrMemoryPoolSupportedHandleTypes,
+                                                   rmm::detail::current_device().value());
 
-    // Don't throw on cudaErrorInvalidValue
-    auto const unsupported_runtime = (result == cudaErrorInvalidValue);
-    if (unsupported_runtime) return false;
-    // throw any other error that may have occurred
-    RMM_CUDA_TRY(result);
+        // Don't throw on cudaErrorInvalidValue
+        auto const unsupported_runtime = (result == cudaErrorInvalidValue);
+        if (unsupported_runtime) return false;
+        // throw any other error that may have occurred
+        RMM_CUDA_TRY(result);
+    }
 
 #endif
     return (supported_handle_types_bitmask & handle_type) == handle_type;

--- a/include/rmm/detail/dynamic_load_runtime.hpp
+++ b/include/rmm/detail/dynamic_load_runtime.hpp
@@ -136,9 +136,16 @@ struct async_alloc {
   {
     int supported_handle_types_bitmask{};
 #if CUDART_VERSION >= 11030  // 11.3 introduced cudaDevAttrMemoryPoolSupportedHandleTypes
-    RMM_CUDA_TRY(cudaDeviceGetAttribute(&supported_handle_types_bitmask,
+    auto const result = cudaDeviceGetAttribute(&supported_handle_types_bitmask,
                                         cudaDevAttrMemoryPoolSupportedHandleTypes,
-                                        rmm::detail::current_device().value()));
+                                        rmm::detail::current_device().value());
+
+   // Don't throw on cudaErrorInvalidValue
+   auto const unsupported_runtime = (result == cudaErrorInvalidValue);
+   if (unsupported_runtime) return false;
+   // throw any other error that may have occurred
+   RMM_CUDA_TRY(result);
+
 #endif
     return (supported_handle_types_bitmask & handle_type) == handle_type;
   }

--- a/include/rmm/detail/dynamic_load_runtime.hpp
+++ b/include/rmm/detail/dynamic_load_runtime.hpp
@@ -137,14 +137,14 @@ struct async_alloc {
     int supported_handle_types_bitmask{};
 #if CUDART_VERSION >= 11030  // 11.3 introduced cudaDevAttrMemoryPoolSupportedHandleTypes
     auto const result = cudaDeviceGetAttribute(&supported_handle_types_bitmask,
-                                        cudaDevAttrMemoryPoolSupportedHandleTypes,
-                                        rmm::detail::current_device().value());
+                                               cudaDevAttrMemoryPoolSupportedHandleTypes,
+                                               rmm::detail::current_device().value());
 
-   // Don't throw on cudaErrorInvalidValue
-   auto const unsupported_runtime = (result == cudaErrorInvalidValue);
-   if (unsupported_runtime) return false;
-   // throw any other error that may have occurred
-   RMM_CUDA_TRY(result);
+    // Don't throw on cudaErrorInvalidValue
+    auto const unsupported_runtime = (result == cudaErrorInvalidValue);
+    if (unsupported_runtime) return false;
+    // throw any other error that may have occurred
+    RMM_CUDA_TRY(result);
 
 #endif
     return (supported_handle_types_bitmask & handle_type) == handle_type;

